driver/src/main/com/mongodb/client/GridFSDownloadStream.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client;
+
+import org.bson.BsonDocument;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * A GridFS InputStream for downloading data from GridFS
+ *
+ * @since 3.1
+ */
+public interface GridFSDownloadStream {
```

Comment:
I see why you did it this way--so as not to expose any concrete sub-classes.  But I'd much prefer this:

    public abstract class GridFSDownloadStream extends InputStream {
        public abstract BsonDocument getFileInformation();
    }